README.rst

```diff
@@ -1874,823 +67,14 @@ as in the example below:
-       )
-
-Instead, we represent a *compound* optional by taking an union
-(:code:`UNIONALL`) of two distinct :code:`MATCH` queries. For instance, the
-:code:`GraphQL` query above can be represented as follows:
-
-::
-
-    SELECT EXPAND($final_match)
-    LET
-        $match1 = (
-            SELECT
-                Animal___1.name AS `name`
-            FROM (
-                MATCH {
-                    class: Animal,
-                    as: Animal___1,
-                    where: (
-                        (in_Animal_ParentOf IS null)
-                        OR
-                        (in_Animal_ParentOf.size() = 0)
-                    ),
-                }
-            )
-        ),
-        $match2 = (
-            SELECT
-                Animal___1.name AS `name`,
-                Animal__in_Animal_ParentOf___1.name AS `parent_name`
-            FROM (
-                MATCH {
-                    class: Animal,
-                    as: Animal___1
-                }.in('Animal_ParentOf') {
-                    as: Animal__in_Animal_ParentOf___1
-                }.in('Animal_ParentOf') {
-                    as: Animal__in_Animal_ParentOf__in_Animal_ParentOf___1
-                }
-            )
-        ),
-        $final_match = UNIONALL($match1, $match2)
-
-In the first case where the optional edge is not followed, we have to
-explicitly filter out all vertices where the edge *could have been
-followed*. This is to eliminate duplicates between the two :code:`MATCH`
-selections.
-
-The previous example is not *exactly* how we implement *compound*
-optionals (we also have :code:`SELECT` statements within :code:`$match1` and
-:code:`$match2`), but it illustrates the the general idea.
-
-Performance Penalty
-^^^^^^^^^^^^^^^^^^^
-
-If we have many *compound* optionals in the given :code:`GraphQL`, the above
-procedure results in the union of a large number of :code:`MATCH` queries.
-Specifically, for :code:`n` compound optionals, we generate 2n different
-:code:`MATCH` queries. For each of the 2n subsets :code:`S` of the :code:`n`
-optional edges:
-
-- We remove the :code:`@optional` restriction for each traversal in :code:`S`.
-- For each traverse :code:`t` in the complement of :code:`S`, we entirely discard :code:`t` along
-  with all the vertices and directives within it, and we add a filter on the previous traverse to
-  ensure that the edge corresponding to :code:`t` does not exist.
-
-Therefore, we get a performance penalty that grows exponentially with
-the number of *compound* optional edges. This is important to keep in
-mind when writing queries with many optional directives.
-
-If some of those *compound* optionals contain :code:`@optional` vertex
-fields of their own, the performance penalty grows since we have to
-account for all possible subsets of :code:`@optional` statements that can be
-satisfied simultaneously.
-
-Optional :code:`type_equivalence_hints` parameter
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Comment:
I replaced this section with the more generic: https://graphql-compiler.readthedocs.io/en/latest/language_specification/schema_types.html#union-types-and-type-equivalence-hints

I did fail to explain that the type equivalence hints lets the compiler optimize away type coercions in Gremlin and that Gremlin does not have first-class support for inheritance. This section though does not fit nicely into the ReadTheDocs layout so it will take me sometime to figure out where to put it and gremlin is not really used and we want to deprecate the type equivalence hints parameter. So do you think it's worth spending time on this?

Reply:
I discussed this with Bojan and thought it would be best just to delete this section altogether instead of putting in the work to find where it could nicely fit in ReadTheDocs.